### ISSUE TITLE
Reporter doesnt honor when layer visibility is turned off in webmap

### ISSUE THREAD

**Issue by skrishn** [Esri/solutions-components#751]
### Describe the bug

All the layer's visibility is set to off in the web map. I see the following image on the config panel ( not acceptable layers message), but all the layers appear in the report panel. On load, the map doesn't render features, but when clicked on a layer, all features render and when clicked on a feature on the panel or on the map, it shows the popup in the panel.

### Expected Behavior

The app should honor layer visibility set in webmap for any or all layers. When layer visibility is off, it shouldnt show it in the map or panel. 

### Reproduction Steps & Sample

![image](https://github.com/user-attachments/assets/092814a5-d693-4cdd-a42b-6815fe0891f3)

https://localgovdev.mapsdevext.arcgis.com/apps/instant/reporter/index.html?appid=97f92d8f17ef4509a6a929f2ce863d22

**Comment by skrishn** [comment]
@sumitzarkar The message - non acceptable layer is part of another issue that sporadically happens in the reporter app and we're tracking it separately https://devtopia.esri.com/WebGIS/arcgis-portal-app-templates/issues/10441. As I can see now, the config panel doesn't seem to honor layer visibility and if the webmap has valid editable layers, it shows them in config. After getting the use-case from the user, let's see how to resolve this.

**Comment by skrishn** [comment]
@sumitzarkar We decided to have the same experience as other IA apps, i.e. not rendering on the map but the user can still interact with it on the panel. If the report incident button is enabled, users can add the feature and it will show a feature being added on the map, and once the creation process is complete, it will not render on the map.

**Comment by sumitzarkar** [comment]
@skrishn 

## Cause
- Thi is as designed in previous versions.
- We had a functionality in reporter when we have multiple layers and we select a reporting layer, then we turn off the visibility of all other reporting layers, and againg navigating back to layers list will show all the reporting layers. Due to this when we are navigating back we are turning on the layers visibility.
___

## Solution
 - Don't render the layer on the map but allow its interaction on the panel
 - If the report incident button is enabled, users can add the feature and it will show a feature being added on the map, and once the creation process is complete, it will not render on the map.
 - Get all the layers id from the config and create a array of default visibility for all the reporting layers
 - When turning off all the layers while navigating to features list turnOff only those layers which are visbile and while showing all layer when returning to layerlist turn on only those layers which are on by default.
___

## Dev Checklist


Items ​ | Is required​ | Status | Comments​
-- | -- | -- | --
Required NLS changes?​ | No   |   | ​
Required Backward compatibility​ | No  |   | ​
Required 508​ | No  |   | ​
Need Sanitize?​ | No  |   | ​
___

## Impact Analysis Report

Impacted Areas | Comments
-- | --
Have Configuration changes?​ | No
Have Runtime changes?​ | Yes
Impact on RTL | No
Impacted areas | Test with different cases in which reporting Layers visibility is on or off
___
